erpnext/public/js/address.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2016, Frappe Technologies and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Address", {
+	is_your_company_address: function(frm) {
+		if(frm.doc.is_your_company_address){
+			frm.add_child('links', {
+				link_doctype: 'Company',
+				link_name: frappe.defaults.get_user_default('Company')
+			});
+			frm.fields_dict.links.grid.get_field('link_doctype').get_query = function() {
```

Comment:
A better way to avoid this duplication is to write this method in `onload` event so it will run only once, but it will get the filters dynamically based on form values.

```js
onload(frm) {
    frm.set_query('link_doctype', 'links', () => {
		return {
			filters: frm.doc.is_your_company_address ? {
				name: 'Company'
			} : null
		}
    })
}

```

Reply:
I tried using the onload event but there are [filters being set in address.js](https://github.com/frappe/frappe/blob/develop/frappe/contacts/doctype/address/address.js#L17) in frappe on refresh event which are replacing the one we set with onload, unfortunately.

On taking a closer look I realised that these filters are to list only doctypes in the link_doctype field which have an address field which is actually important. I don't think we can override the refresh event as it has a lot of other code too.

Instead I have replaced the part in address.js in ERPNext where I was setting the field filter to null and triggered a form refresh so that the default filters are reset. Also updated the old query code as you showed.

I have triggered the refresh event using frm.trigger() instead of calling frm.refresh() because the page scrolls up and it's slightly disorienting.

Are these changes fine?